x-pack/plugins/fleet/server/services/package_policies/simplified_package_policy_helper.ts

```diff
@@ -0,0 +1,134 @@
+import type {
+  NewPackagePolicyInput,
+  NewPackagePolicyInputStream,
+  PackagePolicyConfigRecord,
+} from '../../../common/types';
+import type { NewPackagePolicy, PackageInfo } from '../../types';
+
+type SimplifiedVars = Record<string, string | string[] | boolean | number | number[] | null>;
+
+export interface SimplifiedPackagePolicy {
+  id?: string;
+  policy_id: string;
+  namespace: string;
+  name: string;
+  description?: string;
+  vars?: SimplifiedVars;
+  inputs?: Record<
+    string,
+    {
+      enabled?: boolean | undefined;
+      vars?: SimplifiedVars;
+      streams?: Record<
+        string,
+        {
+          enabled?: undefined | boolean;
+          vars?: SimplifiedVars;
+        }
+      >;
+    }
+  >;
+}
+
+export function generateInputId(input: NewPackagePolicyInput) {
+  return `${input.policy_template ? `${input.policy_template}-` : ''}${input.type}`;
+}
+
+function assignVariables(
+  userProvidedVars: SimplifiedVars,
+  varsRecord?: PackagePolicyConfigRecord,
+  ctxMessage = ''
+) {
+  Object.entries(userProvidedVars).forEach(([varKey, varValue]) => {
+    if (!varsRecord || !varsRecord[varKey]) {
+      throw new Error(`Variable ${ctxMessage}:${varKey} not found`);
+    }
+
+    varsRecord[varKey].value = varValue;
+  });
+}
+
+type StreamsMap = Map<string, NewPackagePolicyInputStream>;
+type InputMap = Map<string, { input: NewPackagePolicyInput; streams: StreamsMap }>;
+
+/**
+ *
+ * @param data
+ * @param packageInfo
+ */
+export function simplifiedPackagePolicytoNewPackagePolicy(
+  data: SimplifiedPackagePolicy,
+  packageInfo: PackageInfo
+): NewPackagePolicy {
+  const {
+    policy_id: policyId,
+    namespace,
+    name,
+    description,
+    inputs = {},
+    vars: packageLevelVars,
+  } = data;
+  const packagePolicy = packageToPackagePolicy(packageInfo, policyId, namespace, name, description);
+
+  // Build a input and streams Map to easily find package policy stream
+  const inputMap: InputMap = new Map();
+  // disable everything or use default?
+  packagePolicy.inputs.forEach((input) => {
+    const streamMap: StreamsMap = new Map();
+    input.streams.forEach((stream) => {
+      streamMap.set(stream.data_stream.dataset, stream);
+    });
+    inputMap.set(generateInputId(input), { input, streams: streamMap });
+  });
+
+  if (packageLevelVars) {
+    assignVariables(packageLevelVars, packagePolicy.vars);
+  }
+
+  Object.entries(inputs).forEach(([inputId, val]) => {
+    const { enabled, streams = {}, vars: inputLevelVars } = val;
+
+    const { input: packagePolicyInput, streams: streamsMap } = inputMap.get(inputId) ?? {};
+    if (!packagePolicyInput || !streamsMap) {
+      throw new Error(`Input not found: ${inputId}`);
+    }
+
+    if (enabled === false) {
+      packagePolicyInput.enabled = false;
+    } else {
+      packagePolicyInput.enabled = true;
+    }
+
+    if (inputLevelVars) {
+      assignVariables(inputLevelVars, packagePolicyInput.vars, `${inputId}`);
+    }
```

Comment:
Really like that we have this same `assignVariables` helper being called at each "level" of variable. Makes the code super easy to follow.